packages/firebase_messaging/firebase_messaging_platform_interface/lib/src/method_channel/method_channel_messaging.dart

```diff
@@ -34,78 +30,81 @@ void _firebaseMessagingCallbackDispatcher() {
 
   // This is where we handle background events from the native portion of the plugin.
   _channel.setMethodCallHandler((MethodCall call) async {
-    if (call.method == "MessagingBackground#onMessage") {
+    if (call.method == 'MessagingBackground#onMessage') {
       final CallbackHandle handle =
-          CallbackHandle.fromRawHandle(call.arguments["userCallbackHandle"]);
+          CallbackHandle.fromRawHandle(call.arguments['userCallbackHandle']);
 
       // PluginUtilities.getCallbackFromHandle performs a lookup based on the
       // callback handle and returns a tear-off of the original callback.
-      final Function closure = PluginUtilities.getCallbackFromHandle(handle);
+      final Function closure = PluginUtilities.getCallbackFromHandle(handle)!;
```

Comment:
the type `Function` is scary here

It should be something more specific, like `Future<void> Function(RemoteMessage)`